### PR TITLE
Confirm that modules are always tidy and vendored.

### PR DESCRIPTION
**Pull request by matthewdale**
Confirm that PRs always have "tidy" and correctly vendored modules by running `go mod tidy` and `go mod vendor` during Evergreen CI runs. If `go mod tidy` or `go mod vendor` make any changes, return an error code.

### CODE REVIEW

**Review by benjirewis — APPROVED** [review]
Great addition! LGTM assuming `static-analysis` passes.

**Review by prestonvasquez — APPROVED** [review]

**Comment by matthewdale** [comment]
Currently depends on https://github.com/mongodb/mongo-go-driver/pull/1005. Will merge after that is merged.

**Pull request merged**
